### PR TITLE
Add Microsoft Teams for Linux (Electron) profile

### PR DESCRIPTION
**Pull request by TheDarkTrumpet**

### CODE REVIEW

File: etc/teams-for-linux.profile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
+
+private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh,teams-for-linux
+private-dev
+private-etc fonts,machine-id,localtime,ld.so.cache,ca-certificates,ssl,pki,crypto-policies,resolv.conf
+private-tmp
+disable-mnt
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Move over private-bin.

File: etc/teams-for-linux.profile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+include disable-common.inc
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Add over the include block.
```
# Firejail profile for teams-for-linux
# Description: TODO
# This file is overwritten after every install/update
# Persistent local customizations
include teams-for-linux.local
# Persistent global definitions
include globals.local
```

File: etc/teams-for-linux.profile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+protocol unix,inet,inet6,netlink
+seccomp
+
+private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh,teams-for-linux
+private-dev
+private-etc fonts,machine-id,localtime,ld.so.cache,ca-certificates,ssl,pki,crypto-policies,resolv.conf
+private-tmp
+disable-mnt
+
+noblacklist ${HOME}/.config/teams-for-linux
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Before the includes

File: etc/teams-for-linux.profile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+whitelist ${HOME}/.config/teams-for-linux
+noexec /tmp
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: remove and add `include disable-exec.inc` to the include block.

File: etc/teams-for-linux.profile [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+seccomp
+
+private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh,teams-for-linux
+private-dev
+private-etc fonts,machine-id,localtime,ld.so.cache,ca-certificates,ssl,pki,crypto-policies,resolv.conf
+private-tmp
+disable-mnt
+
+noblacklist ${HOME}/.config/teams-for-linux
+whitelist ${HOME}/.config/teams-for-linux
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Move between `caps.drop all` and the includes.

File: etc/teams-for-linux.profile [hunk 6]
```diff
@@ -0,0 +1,27 @@
+include disable-common.inc
+include disable-devel.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include globals.local
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: remove here

File: etc/teams-for-linux.profile [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
+
+private-bin sh,xdg-mime,tr,sed,echo,head,cut,xdg-open,grep,egrep,bash,zsh,teams-for-linux
+private-dev
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: What about `private-cache`?

**Comment by rusty-snake** [comment]
Also add `teams-for-linux` to `src/firecfg/firecfg.config`.

File: etc/teams-for-linux.profile [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+netfilter
+nodvd
+nogroups
+nonewprivs
+noroot
+notv
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: What about `shell none` and `tracelog`?

File: etc/teams-for-linux.profile [hunk 9]
```diff
@@ -0,0 +1,27 @@
+include disable-common.inc
+include disable-devel.inc
+include disable-passwdmgr.inc
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Add `include disable-interpreters.inc`

File: etc/teams-for-linux.profile [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+include disable-programs.inc
+include globals.local
+
+caps.drop all
+netfilter
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Need it dbus access or can we add `nodbus`?

Comment (TheDarkTrumpet) [reply etc/teams-for-linux.profile]: It'll need dbus for notification ability. I'll add the other suggestions, and test them out on my install before updating the pull request. Thank you for the feedback, first time contributing to this, and I appreciate greatly the suggestions.

**Comment by Fred-Barclay** [comment]
@TheDarkTrumpet any updates on this? :smile:

**Comment by rusty-snake** [comment]
@Fred-Barclay note his status :wink: : I may be slow to respond.

**Comment by TheDarkTrumpet** [comment]
heh, thanks for the poke on that.  Sorry for not responding earlier - starting a new job soon so work has been pretty crazy right now trying to wrap up stuff in current job and prepare for the new one.  I have time this weekend to implement the changes and verify.  I'll have them pushed by the end of this weekend

File: etc/teams-for-linux.profile (hunk 8 above)
Comment (TheDarkTrumpet) [reply etc/teams-for-linux.profile]: Tried adding both of these, and the program doesn't load with these defined (both).  Curious about what tracelog does, so I'll grep the source to learn about that.

**Comment by Fred-Barclay** [comment]
@TheDarkTrumpet Thanks! No rush, just was wondering if there was an update. :smiley: Congrats on the new job!

@rusty-snake Ah yes, got to remember we have that feature on GitHub now..

File: etc/teams-for-linux.profile (hunk 8 above)
Comment (rusty-snake) [reply etc/teams-for-linux.profile]: If it break, leave it. If you don't know what option XY do, you can read the source-code or read the manpages (`man firejail` and `man firejail-profile`) :wink:

Comment (TheDarkTrumpet) [reply etc/teams-for-linux.profile]: I was able to get shell none to work, but not tracelog.  Committed and pushed that part of the change.

File: etc/teams-for-linux.profile (hunk 1 above)
Comment (TheDarkTrumpet) [reply etc/teams-for-linux.profile]: This was the only comment I wasn't sure about what was meant.  When you mention move over private-bin, what do you mean?  Thanks again for all your help in looking at this.

**Comment by TheDarkTrumpet** [comment]
Alright, I think I got all the changes added minus the "move over private-bin" part.  Please let me know if there's anything I missed or anything you suggest I try adding.  I'll probably spend some more time adding new profiles for other programs I use as there's a few that could benefit from this.  Thanks for putting together this fantastic project.

**Comment by SkewedZeppelin** [comment]
@TheDarkTrumpet 

> move over private-bin

https://gist.github.com/SkewedZeppelin/f85777e5094e8ff899d5b7346bda2b8a

File: etc/teams-for-linux.profile (hunk 1 above)
Comment (rusty-snake) [reply etc/teams-for-linux.profile]: This order:
```
disable-mnt
private-bin
private-cache
private-dev
private-etc
private-tmp
```

File: etc/teams-for-linux.profile [hunk 11]
```diff
@@ -0,0 +1,43 @@
+# Firejail profile for teams-for-linux
+# Description: Teams for Linux is an Electron application for Microsoft's team collaboration and chat program
+# This file is overwritten after every install/update
+# Persistent local customizations
+include teams-for-linux.local
+# Persistent global definitions
+include globals.local
+
+caps.drop all
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Move down, over `netfilter`

File: etc/teams-for-linux.profile [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+mkdir ${HOME}/.config/teams-for-linux
+whitelist ${HOME}/.config/teams-for-linux
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Move `mkdir` and `whitelist` down.

File: etc/teams-for-linux.profile [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+include whitelist-common.inc
+include whitelist-var-common.inc
+include disable-common.inc
+include disable-devel.inc
+include disable-passwdmgr.inc
+include disable-interpreters.inc
+include disable-exec.inc
+include disable-programs.inc
+
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]:
```suggestion

mkdir ${HOME}/.config/teams-for-linux
whitelist ${HOME}/.config/teams-for-linux
include whitelist-common.inc
include whitelist-var-common.inc

```

File: etc/teams-for-linux.profile [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+include globals.local
+
+caps.drop all
+
+mkdir ${HOME}/.config/teams-for-linux
+whitelist ${HOME}/.config/teams-for-linux
+noblacklist ${HOME}/.config/teams-for-linux
+
+include whitelist-common.inc
+include whitelist-var-common.inc
```
Comment (rusty-snake) [inline etc/teams-for-linux.profile]: Move down.

**Comment by rusty-snake** [comment]
Uhh I see @SkewedZeppelin has already posted a patch.

**Comment by TheDarkTrumpet** [comment]
I pushed the changes according to the patch (I didn't apply it directly since some was already added).

I'm thinking a bit, and I'm curious if the https://github.com/netblue30/firejail/blob/master/CONTRIBUTING.md can be improved a bit.  At first, I could see the improvements from a security standpoint, but quickly became more about making things alphabetical and grouped in certain ways.  I definitely understand the desire to keep things clean though.  While I feel like I got a decent handle on the desires at this point, it wasn't until the diff popped up that I noticed the alphabetical nature of it all.  It probably didn't help I used Discord as a jumping off point for Teams since they're similar in terms of functionality.

I do want to help with other profiles, and I want to see others also help with the profiles.  I think updating the Contributing that kinda spells out at least the formatting desires would be great, then the bulk of the pull requests could be better served in adding and tightening security concerns.

**Comment by SkewedZeppelin** [comment]
This looks fine now.
Merged, thanks.

As for the format and new contributor issues, they are very real.
But many have been working towards improving it.
See the recently added template
https://github.com/netblue30/firejail/blob/master/etc/templates/profile.template
and the goals on improving documentation and a wiki in #2729

**Comment by TheDarkTrumpet** [comment]
That profile.template is amazingly helpful.  Wish I found that earlier.

I found another issue (#2730 ) that I commented on as well. Going to subscribe to both issues.  The tl;dr for pretty much anything I can recommend is the Table of Contents, or the jumping off point, should be bubbled up to the most likely place people would start looking.  Contrib, or readme.  I'll mention the same thing in #2729 to keep issue discussion with issues.

**Pull request merged**